### PR TITLE
Snapshot test for usernameToColor function

### PR DESCRIPTION
**Pull request by juliushaertl**
This adds some basic snapshot testing for the usernameToColor functionality, which mainly aims to get a testset that we can use to test different implementations of the same algorithm against, e.g. in the android app.

A JSON export can be generated with 
`node ./build/usernameToColor-export.js`

<details>
<summary>JSON output for the current snapshot</summary>

```{"":"#0082c9",",":"#1e78c1",".":"#c98879","123e4567-e89b-12d3-a456-426614174000":"#bc5c91","Akeel Robertson":"#9750a4","Alishia Ann Lowry":"#d09e6d","Arham Johnson":"#0082c9","Brayden Truong":"#d09e6d","Daphne Roy":"#9750a4","Ellena Wright Frederic Conway":"#c37285","Gianluca Hills":"#d6b461","Haseeb Stephens":"#d6b461","Idris Mac":"#9750a4","Kristi Fisher":"#0082c9","Lillian Wall":"#bc5c91","Lorelai Taylor":"#ddcb55","Madina Knight":"#9750a4","Meeting":"#c98879","Private Circle":"#c37285","Rae Hope":"#795aab","Santiago Singleton":"#bc5c91","Sid Combs":"#d09e6d","TestCircle":"#499aa2","Tom Mörtel":"#248eb5","Vivienne Jacobs":"#1e78c1","Zaki Cortes":"#6ea68f","a user":"#5b64b3","admin":"#d09e6d","admin@cloud.example.com":"#9750a4","another user":"#ddcb55","asd":"#248eb5","bar":"#0082c9","foo":"#d09e6d","wasd":"#b6469d","مرحبا بالعالم":"#c98879","🙈":"#b6469d"}```

</details>

cc @tobiasKaminsky

### CODE REVIEW

File: build/usernameToColor-export.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const snapshot = require('./../src/functions/usernameToColor/__snapshots__/usernameToColor.spec.js.snap')
+const result = {}
+Object.keys(snapshot).map((key) => {
+		const uid = key.replace('usernameToColor ', '').replace(' has the proper color 1', '')
```
Comment (juliushaertl) [inline build/usernameToColor-export.js:4]: There might be a more sane way to properly deserialize the jest snapshot but for now this does the trick

**Comment by tobiasKaminsky** [comment]
Can we also have a tests for
- "" (no name)
- Meeting
- Private Circle
- TestCircle
(just so that we have evry share type included)

File: src/functions/usernameToColor/__snapshots__/usernameToColor.spec.js.snap [hunk 2]
```diff
@@ -0,0 +1,209 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`usernameToColor Akeel Robertson has the proper color 1`] = `
+Color {
+  "b": 164,
```
Comment (tobiasKaminsky) [inline src/functions/usernameToColor/__snapshots__/usernameToColor.spec.js.snap:5]: That is hard to read, as normally one would expect r,g,b.

Comment (juliushaertl) [reply src/functions/usernameToColor/__snapshots__/usernameToColor.spec.js.snap:5]: Yeah, we can probably just convert it to hex for the test and snapshot

Comment (juliushaertl) [reply src/functions/usernameToColor/__snapshots__/usernameToColor.spec.js.snap]: Done and updated the json in the first post.

**Comment by skjnldsv** [comment]
Ping? :D

**Comment by juliushaertl** [comment]
Pong ....

Waiting for reviews :grin:

File: src/functions/usernameToColor/usernameToColor.spec.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import usernameToColor from './index'
```
Comment (skjnldsv) [inline src/functions/usernameToColor/usernameToColor.spec.js:1]: Quick note though, wouldn't it better to put them in /tests ?

Comment (skjnldsv) [reply src/functions/usernameToColor/usernameToColor.spec.js:1]: @juliushaertl

Comment (juliushaertl) [reply src/functions/usernameToColor/usernameToColor.spec.js]: Definitely, pushed a commit for that.

**Review by skjnldsv — APPROVED** [review]
Still :ok_hand:

**Review by ma12-co — APPROVED** [review]
Tested and works!

**Pull request merged**